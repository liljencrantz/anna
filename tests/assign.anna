
def main(){
    print("Int\n");
    var i = 3;
    i++;
    print(i, "\n");
    print(i--, "\n");
    i+= 3;
    print(i, "\n");
    i-= 1;
    print(i, "\n");

    print("Float\n");
    var f = 0.0;
    f += 5;
    print(f, "\n");

    print("Char\n");
    var c = 'a';
    c += 3;
    print(c, "\n");

    print("List\n");
    var l = [1];
    print(l.count, l[0], l[1], "\n");
    l ~= [5];
    print(l.count, l[0], l[1], "\n");

    print("String\n");
    var s = "aaa";
    s ~= "bbb";
    s ~= 'c';
    print(s, "\n");

}
