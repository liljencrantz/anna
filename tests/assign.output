Int
4
3
6
5
Float
5.000000
Char
d
List
1 1 null
2 1 5
String
aaabbbc
