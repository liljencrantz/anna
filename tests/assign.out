Int
4
3
6
5
Float
5.000000
Char
d
List
11null
215
String
aaabbbc
